test/e2e/performanceprofile/functests/6_mustgather_testing/mustgather.go

```diff
@@ -0,0 +1,178 @@
+			}
+		}
+
+		// pre generated must-gather data
+		// hardcored node name
+		// TODO genearate data in live cluster
+		snapshotDir = filepath.Join(mustGatherContentDir, "nodes", "ci-ln-t0prq3t-72292-h4x2n-worker-a-wqnqt")
+		snapshotName = filepath.Join(snapshotDir, "sysinfo.tgz")
+
+		if _, err := os.Stat(snapshotName); errors.Is(err, os.ErrNotExist) {
+			Expect(err).To(BeNil(), "failed to read sysinfo.tgz file %s: %v", snapshotName, err)
+		}
+
+		err = Untar(snapshotDir, snapshotName)
+		Expect(err).To(BeNil(), "failed to read the %s: %v", snapshotName, err)
+	})
+	Context("PAO must-gather Tests", func() {
+		It("Verify Generic cluster resource definitions are captured", func() {
+			var genericFiles = []string{
+				"version",
+				"cluster-scoped-resources/config.openshift.io/featuregates/cluster.yaml",
+				"namespaces/openshift-cluster-node-tuning-operator/tuned.openshift.io/tuneds/default.yaml",
+				"namespaces/openshift-cluster-node-tuning-operator/tuned.openshift.io/tuneds/rendered.yaml",
+			}
+			err := checkfilesExist(genericFiles, mustGatherContentDir)
+			Expect(err).To(BeNil())
+		})
+
+		It("Verify PAO cluster resources are captured", func() {
+			profile, _ = profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+			if profile != nil {
+				//replace peformance.yaml for profile.Name when data is generated in the node
+				ClusterSpecificFiles := []string{
+					"cluster-scoped-resources/performance.openshift.io/performanceprofiles/performance.yaml",
+					"cluster-scoped-resources/machineconfiguration.openshift.io/kubeletconfigs/performance-performance.yaml",
+					"namespaces/openshift-cluster-node-tuning-operator/tuned.openshift.io/tuneds/openshift-node-performance-performance.yaml",
+				}
+				err := checkfilesExist(ClusterSpecificFiles, mustGatherContentDir)
+				Expect(err).To(BeNil())
+			} else {
+				Skip("No Performance Profile found")
+			}
+		})
+		It("Verify hardware related information are captured", func() {
+			nodeSpecificFiles := []string{
+				"sys/devices/system/cpu/offline",
+				"proc/cpuinfo",
+				"cpu_affinities.json",
+				"dmesg",
+				"irq_affinities.json",
+				"lscpu",
+				"machineinfo.json",
+				"podresources.json",
+				"proc_cmdline",
+				"sysinfo.log",
+			}
+			err := checkfilesExist(nodeSpecificFiles, snapshotDir)
+			Expect(err).To(BeNil())
+
+		})
+		It("Verify machineconfig resources are captured", func() {
+			mcps := &machineconfigv1.MachineConfigPoolList{}
+			selector := labels.NewSelector()
+			err := testclient.Client.List(context.TODO(), mcps, &client.ListOptions{LabelSelector: selector})
+			Expect(err).ToNot(HaveOccurred())
+			mcpFiles := make([]string, len(mcps.Items))
+			for _, item := range mcps.Items {
+				mcpFiles = append(mcpFiles, fmt.Sprintf("cluster-scoped-resources/machineconfiguration.openshift.io/machineconfigpools/%s.yaml", item.Name))
+			}
+			err = checkfilesExist(mcpFiles, mustGatherContentDir)
```

Comment:
will checkfilesExist return an error if there is one ? or will ginkgo exit in the function itself ?

Reply:
Ginkgo launched and error, but now is impreoved.